frontend/src/routes/(authenticated)/admin/EditUserAccount.svelte

```diff
@@ -16,15 +16,23 @@
   export let currUser: LexAuthUser;
   export let deleteUser: (user: User) => void;
 
-  const schema = z.object({
+  const schema = z
+    .object({
     email: z.string().email($t('form.invalid_email')).nullish(),
     name: z.string(),
     password: passwordFormRules($t).or(emptyString()).default(''),
     score: z.number(),
     role: z.enum([UserRole.User, UserRole.Admin]),
-  });
+    })
+  const refinedSchema = schema
+    .refine((data) => data.email || data.role !== UserRole.Admin, {
```

Comment:
this should also require the email to be verified, something like this should work `(data.email || data.role !== UserRole.Admin) && _user.emailVerified`
Then update the text to specify that the user must provide a verified email address.

Reply:
I'm not super happy closing over `_user` in the the `.refine()` call here, because I believe it will create a bug, at least if my mental model of what happens to Javascript closures when the variable they close over is reassigned is correct. Will the `.refine()` call keep a reference to the old object or the new one? I believe it would keep the old one, i.e. it closes over the current value, rather than over the variable itself. And if so, then there could be a bug where the admin edits one user with a verified email address, then closes the form, then reopens it on another user without a validated email. I believe our EmailModal component is *not* torn down and re-created if the admin stays on the dashboard page, in which case we could end up believing the second user is verified when he really isn't.

The right solution, I think, will be to move the emailVerified value into the `$form`. It doesn't need to have a corresponding HTML input element, but I think it needs to be in the form store so that `.refine()` is only working with values from the form, and not referencing any external objects.

Reply:
I think adding emailVerified to the schema makes sense, it'll need to be passed in to `formModal.open` but that should be pretty simple. I agree I wasn't super comfortable closing over `_user`